tools/node_maintain.py

```diff
@@ -0,0 +1,220 @@
+            host, state = node["nodeHostName"], node["state"]
+            current_nodes[host] = state
+        return current_nodes
+
+    def decommission_nodes(self):
+        try:
+            subprocess.check_output(self.update_command, stderr=subprocess.STDOUT, shell=True)
+        except subprocess.CalledProcessError as e:
+            logger.error(e.output)
+
+
+def get_unready_nodes(decommissioned_nodes, current_status):
+    unready_nodes = {}
+    for node, state in current_status.iteritems():
+        # should decommission but not
+        if state not in {"DECOMMISSIONING", "DECOMMISSIONED"} and node in decommissioned_nodes:
+            unready_nodes[node] = state
+        # should recommission but not
+        if state in {"DECOMMISSIONING", "DECOMMISSIONED"} and node not in decommissioned_nodes:
+            unready_nodes[node] = state
+    return unready_nodes
+
+
+def get_decommission_nodes(args):
+    k8s_operator = KubernetesOperator(args.ip)
+    existing_nodes = k8s_operator.get_nodes()
+    logger.info("Current unhealthy node list: {}".format(','.join(existing_nodes)))
+    return existing_nodes
+
+
+def add_decommission_nodes(args):
+    k8s_operator = KubernetesOperator(args.ip)
+    existing_nodes = k8s_operator.get_nodes()
+    nodes = args.nodes
+    if isinstance(nodes, str):
+        nodes = set(nodes.split(','))
+    full_list = existing_nodes | nodes
+    k8s_operator.set_nodes(full_list)
+    logger.info("Add node: {}".format(args.nodes))
+    return full_list
+
+
+def remove_decommission_nodes(args):
+    k8s_operator = KubernetesOperator(args.ip)
+    existing_nodes = k8s_operator.get_nodes()
+    nodes = args.nodes
+    if isinstance(nodes, str):
+        nodes = set(nodes.split(','))
+    full_list = existing_nodes - nodes
+    k8s_operator.set_nodes(full_list)
+    logger.info("Remove node: {}".format(args.nodes))
+    return full_list
+
+
+def update_decommission_nodes(args):
+    k8s_operator = KubernetesOperator(args.ip)
+    nodes = args.nodes
+    if isinstance(nodes, str):
+        nodes = set(nodes.split(','))
+    k8s_operator.set_nodes(nodes)
+    logger.info("Update node list: {}".format(args.nodes))
+    return nodes
+
+
+def refresh_yarn_nodes(args):
+    k8s_operator = KubernetesOperator(args.ip)
+    yarn_operator = YarnOperator(args.ip)
+    while True:
+        yarn_operator.decommission_nodes()
+        current_status = yarn_operator.get_node_status()
+        decommissioned_nodes = k8s_operator.get_nodes()
+        unready_nodes = get_unready_nodes(decommissioned_nodes, current_status)
+        if len(unready_nodes) == 0:
+            break
+        else:
+            logger.info("Unready nodes: {}".format(unready_nodes))
+        logger.info("Waiting...")
+        time.sleep(30)
+    logger.info("Successfully refresh nodes.")
+
+def setup_parser():
+    top_parser = argparse.ArgumentParser()
+    top_parser.add_argument("ip", help="Master node ip")
```

Comment:
change "Master node ip" to a more meaningful name

Reply:
add more details in parser